CMakeLists.txt

```diff
@@ -23,9 +23,9 @@ endif()
 SET(PLATFORM_LIBRARIES)
 
 if(CMAKE_CXX_STANDARD EQUAL 98)
-   message(FATAL_ERROR "chainbase requires c++17 or newer")
+   message(FATAL_ERROR "chainbase requires c++20 or newer")
 elseif(NOT CMAKE_CXX_STANDARD)
-   set(CMAKE_CXX_STANDARD 17)
+   set(CMAKE_CXX_STANDARD 20)
```

Comment:
Not critical for this PR but something that can be done in the future,

This,
https://github.com/AntelopeIO/chainbase/blob/7817736689d3f8271c3b78c9b5bf8cea6f8b0638/CMakeLists.txt#L2
should be bumped to 3.12 as that's the first version that knows c++20.

Also this entire `if/elseif/endif` block is logically nonsensical. My guess was it originally required c++11, and it would make sense in that case. I might suggest changing the way this is done to how the bls lib does it.

Reply:
Will do!